### PR TITLE
split long lines of js input for node.js to avoid tty issues

### PR DESCRIPTION
**Pull request by joemarshall**
This fixes issue #139 . It would be nice to get it merged so I can test node.js in urllib3

### CODE REVIEW

**Comment by joemarshall** [comment]
Hmm, it seems to break a couple of tests. I'll look at it later.

**Comment by joemarshall** [comment]
Hi @ryanking13 if there's any chance you could look over this - I'd like to merge it soon because it is needed for https://github.com/urllib3/urllib3/pull/3427 to pass testing.

Basically rather than relying on the tty that node is running in to accept super long lines, it instead splits the lines sent to node.js into shorter lines and reassembles them at the node end. 

This is needed because in nox tests in urllib3, it is running inside a subprocess, so the change to the tty doesn't affect the subprocess stdin/out, and instead affects the outer tty.

File: pytest_pyodide/node_test_driver.js [hunk 1]
```diff
@@ -55,7 +55,13 @@ rl.on("line", async function (line) {
     return;
   }
   if (line !== cur_uuid) {
-    cur_code += line + "\n";
+    line = line.substring(0,line.lastIndexOf('$'))
+    if( line === ""){
+      cur_code += "\n";
+    }else{
+      cur_code += line;
+    }
```
Comment (ryanking13) [inline pytest_pyodide/node_test_driver.js:63]: Better formatting:

```suggestion
    line = line.substring(0, line.lastIndexOf('$'))
    if(line === ""){
      cur_code += "\n";
    } else {
      cur_code += line;
    }
```

File: pytest_pyodide/node_test_driver.js [hunk 2]
```diff
@@ -55,7 +55,13 @@ rl.on("line", async function (line) {
...
+    console.log("OK")
```
Comment (ryanking13) [inline pytest_pyodide/node_test_driver.js:64]: If this is a delimiter, let's use a string that is a bit more clear and less likely to be passed by the user.

Also, please add some comment that cross-references the where this value is used.

File: pytest_pyodide/runner.py [hunk 3]
```diff
@@ -610,7 +607,20 @@ def run_js_inner(self, code, check_code):
 
         cmd_id = str(uuid4())
         self.p.sendline(cmd_id)
-        self.p.sendline(wrapped)
+        # split lines into shorter buffers
+        # because some ttys don't like long
```
Comment (ryanking13) [inline pytest_pyodide/runner.py:611]: Could you add a test for passing a very long string? I think we don't have a testcase for this.

File: pytest_pyodide/node_test_driver.js [hunk 4]
```diff
@@ -55,7 +55,13 @@ rl.on("line", async function (line) {
     return;
   }
   if (line !== cur_uuid) {
-    cur_code += line + "\n";
+    line = line.substring(0,line.lastIndexOf('$'))
```
Comment (ryanking13) [inline pytest_pyodide/node_test_driver.js:58]: Similar to my previous comment, please add a comment that where does the delimiter `$` comes from.

File: pytest_pyodide/runner.py [hunk 5]
```diff
@@ -610,7 +607,20 @@ def run_js_inner(self, code, check_code):
...
+        # single lines
+        all_lines = wrapped.split("\n")
+        for c, line in enumerate(all_lines):
+            count = 0
+            while count < len(line):
+                to_read = min(128, len(line) - count)
+                self.p.sendline(line[count : count + to_read] + "$")
```
Comment (ryanking13) [inline pytest_pyodide/runner.py:618]: Here too. Please add a comment about the delimiter `$`

**Review by ryanking13 — APPROVED** [review]
Thanks @joemarshall. I left some minor comments about the readability for future readers of this code. The feature-wise looks good to me. So feel free to merge and release after resolving my comments.

File: pytest_pyodide/runner.py (hunk 3 above)
Comment (joemarshall) [reply pytest_pyodide/runner.py:611]: I think run_in_pyodide tests cover this - they send a 5kb string.

**Pull request merged**